docs/input_output.md

```diff
@@ -2,85 +2,279 @@
-          "config": "<string>",  // The name of the bucket holding configuration and deployment data
-          "private": "<string>", // The name of the bucket which holds internal platform data
-          "public": "<string>"   // The name of the bucket which holds data to be served publicly
-        },
+      "workflow_config": {
+        "ExampleTask": {
+          "cumulus_message": {
+            "outputs": [
+              {
+                "source": "{{$}}",
+                "destination": "{{$.payload}}"
+              },
+              {
+                "source": "{{$.output.anykey}}",
+                "destination": "{{$.meta.baz}}"
+              }
+            ]
+          }
+        }
       },
-      "collection": { ... },   // The full, original collection configuration, as specified in
-                               // collections.json, with inheritance resolved. Tasks should not
-                               // modify this but must pass it on.
-
-      "meta": { ... },         // Metadata about this ingest execution. It is initialized to
-                               // the "meta" attribute of the collection (or the empty object).
-                               // Tasks may add fields to the "meta" object at will (in their
-                               // returned output) in order to pass data to future tasks.
-                               // Tasks should avoid assuming that fields are present in the
-                               // meta object and avoid naming fields to put in the meta object,
-                               // preferring instead to let configuration decide what goes into
-                               // the object. See VIIRS discovery in collections.json for an example.
-                               // It uses a field "addMeta" to determine which parts of a discovered
-                               // URL should be added to the meta object under which fields.
-                               // Subsequent tasks are told to read those fields by using JSON templates.
-
-        "payload": ...         // A Task-specific payload. This can be any data type required by
-                               // the Task. It can be considered the input and output of the Task,
-                               // whereas the other fields are execution context. Tasks should
-                               // document their expected payload input and output formats.
-                               // Generally a Task will return an object which is nearly identical
-                               // to its input in all fields but "payload", and "payload" will be
-                               // completely different
+      "meta": {
+        "foo": "bar",
+        "baz": "boo"
+      },
+      "payload": {
+        "output": {
+          "anykey": "boo"
+        }
+      }
     }
+    
+### 5. Validate task input, output and configuration messages against the schemas provided.
```

Comment:
Should put something here about the default location of the schemas being the schemas folder in the top level of the task and the default filenames being input.json, output.json, and config.json. We're also updating the handlers to take schema locations. 

So if they input a schema location, we'll use that. If not, we'll check the default location. If no schema can be found, then we will not validate.